cmd/apps/rancher_app.go

```diff
@@ -0,0 +1,187 @@
+
+		log.Printf("Client: %s, %s\n", clientArch, clientOS)
+		log.Printf("User dir established as: %s\n", userPath)
+
+		// exit on arm
+		arch := k8s.GetNodeArchitecture()
+		fmt.Printf("Node architecture: %q\n", arch)
+
+		if arch != IntelArch {
+			return fmt.Errorf(`ARM is experimental and not supported by Rancher https://rancher.com/docs/rancher/v2.x/en/installation/options/arm64-platform/`)
+		}
+
+		// create the namespace
+		nsRes, nsErr := k8s.KubectlTask("create", "namespace", namespace)
+		if nsErr != nil {
+			return nsErr
+		}
+
+		// ignore errors
+		if nsRes.ExitCode != 0 {
+			log.Printf("[Warning] unable to create namespace %s, may already exist: %s", namespace, nsRes.Stderr)
+		}
+
+		// Deploy TLS cert secret if given
+		if tlsCert != "" || tlsKey != "" {
+			certOption := fmt.Sprintf("--cert=%s", tlsCert)
+			keyOption := fmt.Sprintf("--key=%s", tlsKey)
+			tlsRes, tlsErr := k8s.KubectlTask("--namespace", namespace, "create", "secret", "tls", "tls-rancher-ingress", certOption, keyOption)
+			if tlsErr != nil {
+				return tlsErr
+			}
+
+			if tlsRes.ExitCode != 0 {
+				log.Printf("[Warning] unable to create tls secret for rancher, you will have to do it manually later")
+			}
+		}
+
+		if tlsCa != "" {
+			caOption := fmt.Sprintf("--from-file=cacerts.pem=%s", tlsCa)
+
+			caRes, caErr := k8s.KubectlTask("--namespace", namespace, "create", "secret", "generic", "tls-ca", caOption)
+			if caErr != nil {
+				return caErr
+			}
+
+			if caRes.ExitCode != 0 {
+				log.Printf("[Warning] unable to create tls-ca secret for rancher, you will have to do it manually later")
+			}
+		}
+
+		// set overrides for chart install
+		overrides := map[string]string{}
+		overrides["hostname"] = hostname
+		overrides["replicas"] = fmt.Sprintf("%d", replicas)
+		overrides["ingress.tls.source"] = ingressSource
+
+		if ingressSource == "letsEncrypt" {
+			overrides["letsEncrypt.email"] = letsencryptEmail
+		}
+
+		if tlsCa != "" {
+			overrides["privateCA"] = "true"
+		}
+
+		// install the chart
+		helmHome := path.Join(userPath, ".helm")
+		updateRepo, _ := rancher.Flags().GetBool("update-repo")
+
+		rancherOptions := types.DefaultInstallOptions().
+			WithNamespace(namespace).
+			WithHelmPath(helmHome).
+			WithHelmRepo("rancher-stable/rancher").
+			WithHelmURL("https://releases.rancher.com/server-charts/stable").
+			WithHelmUpdateRepo(updateRepo).
+			WithOverrides(overrides)
+
+		if command.Flags().Changed("kubeconfig") {
+			kubeconfigPath, _ := command.Flags().GetString("kubeconfig")
+			rancherOptions.WithKubeconfigPath(kubeconfigPath)
+		}
+
+		os.Setenv("HELM_HOME", helmHome)
+
+		_, err = helm.TryDownloadHelm(userPath, clientArch, clientOS, true)
+		if err != nil {
+			return err
+		}
+
+		_, err = apps.MakeInstallChart(rancherOptions)
+		if err != nil {
+			return err
+		}
+
+		fmt.Println(rancherInstallMsg)
+
+		return nil
+	}
+
+	return rancher
+}
+
+const RancherInfoMsg = `
+# Wait for Rancher to be rolled out:
```

Comment:
Please could you also add a URL for the docs / getting started page?

Reply:
Added the link to docs